libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -62,14 +71,21 @@ export class MessageRepository extends BaseRepository<MessageDBModel, MessageEnt
       requestQuery.read = { $in: [true, false] };
     }
 
+    if (query.payload) {
+      requestQuery = {
+        ...requestQuery,
+        ...getFlatObject({ payload: query.payload }),
+      };
+    }
```

Comment:
If the payload object is passed, then prepare the flat object out of it, like if the payload is `{ foo: { bar: 123 } }`, then the flat object would be:
```
{
"payload.foo.bar": 123
}
```
Which is passed then to the find query.